Add files via upload

nazarovaaaa opened the pull request:
DeepGraphAnalysis_HW1_Nazarov

nazarovaaaa committed:
Add files via upload

rustam-azimov reviewed (changes requested):
1) Для интереса попробуйте другие генераторы датасета, не такие как на семинаре.
2) Для кратчайших расстояний слишком подробный вектор использовали со всеми парами расстояний. Так не хотим делать, так как даже один и тот же граф после перенумерации вершин даст другой вектор. Хотим получить вектор сколько каких кратчайших длин в графе (по аналогии с тем сколько каких цветов в графе). Размер такого вектора - максимальная кратчайшая длина во всем наборе данных.
3) В WL-kernel можно слегка короче перебирать вершины, полезная функция neighbors.

nazarovaaaa committed:
Add files via upload

Внес изменения согласно комментариям к первой версии файла:
1.Использовал новую функцию генерации: задался задачей классификации графов-звезд и графов-лестниц.
2.Изменил расчет ядра по кратчайшим расстояниям и опробовал функцию на обоих датасетах (на исходном и полученном в п.1).
3.Упростил немного реализацию ядра Weisfeiler-Lehman за счет использования функции neighbors, удалил часть лишних комментариев.

rustam-azimov reviewed (approved):
10 баллов
